### PR TITLE
Anastasiakiseleva patch 1

### PR DESCRIPTION
**Pull request by anastasiakiseleva**

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -1,40 +1,39 @@
...
-<!-- default file list -->
-*Files to look at*:
 
-* [ActionsInvokerViewController.cs](./CS/EFCore/ExecuteActionEF/ExecuteActionEF.Module/Controllers/ActionInvokerViewController.cs) 
-<!-- default file list end -->
-# How to execute Actions in code
+# XAF - How to execute Actions programmatically
 
-## Scenario
+This example explains how to trigger code inside the `Execute` event handlers of custom or standard XAF Actions programmatically, for example, to execute default Action logic that shows a View or runs code in the background.
```
Comment (Abadzhev) [inline Readme.md]:
```suggestion
This example explains how you can trigger the code inside `Execute` event handlers of custom or standard XAF Actions. For example, you can programmatically invoke an Action that shows a View or runs code in the background.
```

File: Readme.md [hunk 2]
```diff
@@ -1,40 +1,39 @@
...
 
-Normally, XAF Actions are executed only by end users using the interaction with UI elements (for instance, button clicks, navigation, or main menu selection). In advanced UI scenarios, XAF developers may want to programmatically trigger the code inside the Execute event handlers of custom or standard XAF Actions. For instance, a XAF developer wants to execute default Action logic that shows a View or runs some code in the background.
+> **NOTE** 
+> We do not recommend this approach. Refactor your Action's `Execute` and other event handlers and extract the required code into separate methods. You can call these methods directly without triggering the UI-related Action code.
```
Comment (Abadzhev) [inline Readme.md]:
```suggestion
> We do not recommend this approach. Refactor your Action's event handlers and extract required code into separate methods. Call these separate methods directly without triggering UI-related Action code.
```

File: Readme.md [hunk 3]
```diff
@@ -1,40 +1,39 @@
...
 
 `mySimpleAction.DoExecute();`
 
 ### ParametrizedAction
 
-`myParametrizedAction.DoExecute("test value"); // Mind the ParametrizedAction.ValueType. Here, a string is used as an example.`
+`myParametrizedAction.DoExecute("test value");`
 
-### SingleChoiceAction
+Mind the `ParametrizedAction.ValueType`. Here, a string is used as an example.
```
Comment (Abadzhev) [inline Readme.md]:
```suggestion
Make sure that the parameter type matches `ParametrizedAction.ValueType`. A string is used in this example.
```

File: Readme.md [hunk 4]
```diff
@@ -1,40 +1,39 @@
...
 
-`mySingleChoiceAction.DoExecute(mySingleChoiceAction.Items[0]); // Mind the SingleChoiceAction.Items structure. Here, the first item is used as an example.`
+### SingleChoiceAction
 
-### PopupWindowShowAction (to display a dialog window)
+`mySingleChoiceAction.DoExecute(mySingleChoiceAction.Items[0]);`
 
-We don't have a public API to execute a PopupWindowShowAction. Use the [ShowViewStrategyBase.ShowViewInPopupWindow(View, Action, Action, String, String)](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.ShowViewStrategyBase.ShowViewInPopupWindow(DevExpress.ExpressApp.View-System.Action-System.Action-System.String-System.String)) method to show a view in a popup.
+Mind the `SingleChoiceAction.Items` structure. Here, the first item is used as an example.
```
Comment (Abadzhev) [inline Readme.md]:
```suggestion
Make sure the pass a parameter that's a valid item from `SingleChoiceAction.Items`. This example passes the first item in the collection.
```

File: Readme.md [hunk 5]
```diff
@@ -1,40 +1,39 @@
...
-<!-- default file list -->
-*Files to look at*:
 
-* [ActionsInvokerViewController.cs](./CS/EFCore/ExecuteActionEF/ExecuteActionEF.Module/Controllers/ActionInvokerViewController.cs) 
-<!-- default file list end -->
-# How to execute Actions in code
+# XAF - How to execute Actions programmatically
 
-## Scenario
+This example explains how you can trigger the code inside `Execute` event handlers of custom or standard XAF Actions. For example, you can programmatically invoke an Action that shows a View or runs code in the background.
```
Comment (dirk-pieterse) [inline Readme.md:8]: This example shows how to trigger the code inside `Execute` event handlers of custom or standard XAF Actions. For example, you can programmatically invoke an Action that shows a View or runs code in the background.

File: Readme.md [hunk 6]
```diff
@@ -1,40 +1,39 @@
...
 
-Normally, XAF Actions are executed only by end users using the interaction with UI elements (for instance, button clicks, navigation, or main menu selection). In advanced UI scenarios, XAF developers may want to programmatically trigger the code inside the Execute event handlers of custom or standard XAF Actions. For instance, a XAF developer wants to execute default Action logic that shows a View or runs some code in the background.
+> **NOTE** 
+> We do not recommend this approach. Refactor your Action's event handlers and extract required code into separate methods. Call these separate methods directly without triggering UI-related Action code.
```
Comment (dirk-pieterse) [inline Readme.md:11]: We do not recommend this approach. Refactor your Action's event handlers and extract the code into separate methods. Call these separate methods directly without triggering UI-related Action code.

File: Readme.md [hunk 7]
```diff
@@ -1,40 +1,39 @@
...
+Make sure that the parameter type matches `ParametrizedAction.ValueType`. A string is used in this example.
 
-`mySingleChoiceAction.DoExecute(mySingleChoiceAction.Items[0]); // Mind the SingleChoiceAction.Items structure. Here, the first item is used as an example.`
+### SingleChoiceAction
 
-### PopupWindowShowAction (to display a dialog window)
+`mySingleChoiceAction.DoExecute(mySingleChoiceAction.Items[0]);`
 
-We don't have a public API to execute a PopupWindowShowAction. Use the [ShowViewStrategyBase.ShowViewInPopupWindow(View, Action, Action, String, String)](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.ShowViewStrategyBase.ShowViewInPopupWindow(DevExpress.ExpressApp.View-System.Action-System.Action-System.String-System.String)) method to show a view in a popup.
+Make sure the pass a parameter that's a valid item from `SingleChoiceAction.Items`. This example passes the first item in the collection.
```
Comment (dirk-pieterse) [inline Readme.md:31]: Make sure you pass a parameter that is a valid item from `SingleChoiceAction.Items`. This example passes the first item in the collection

File: Readme.md (hunk 5 above)
Comment (anastasiakiseleva) [reply Readme.md:8]:
```suggestion
This example shows how to trigger the code inside `Execute` event handlers of custom or standard XAF Actions. For example, you can programmatically invoke an Action that shows a View or runs code in the background.
```

**Pull request merged**